code/modules/clothing/masks/gasmask.dm

```diff
@@ -35,17 +38,29 @@
 	if(LAZYLEN(gas_filters) > 0)
 		. += "<span class='notice'>Currently there [LAZYLEN(gas_filters) == 1 ? "is" : "are"] [LAZYLEN(gas_filters)] filter\s with [get_filter_durability()]% durability.</span>"
 
-/obj/item/clothing/mask/gas/attackby(obj/item/filter, mob/user)
-	if(!istype(filter, /obj/item/gas_filter))
+/obj/item/clothing/mask/gas/attackby(obj/item/tool, mob/user)
+	if(!istype(tool, /obj/item/gas_filter))
 		return ..()
 	if(LAZYLEN(gas_filters) >= max_filters)
 		return ..()
-	if(!user.transferItemToLoc(filter, src))
+	if(!user.transferItemToLoc(tool, src))
 		return ..()
-	LAZYADD(gas_filters, filter)
+	LAZYADD(gas_filters, tool)
 	has_filter = TRUE
 	return TRUE
 
+/obj/item/clothing/mask/gas/attackby_secondary(obj/item/weapon, mob/user, params)
+	if(!has_filter || !max_filters)
+		return ..()
+	for(var/i in 1 to max_filters)
+		var/obj/item/gas_filter/filter = locate() in src
+		if(!filter)
+			continue
+		user.transferItemToLoc(filter, user.loc)
+		LAZYREMOVE(gas_filters, filter)
+	if(LAZYLEN(gas_filters) <= 0)
+		has_filter = FALSE
```

Comment:
Not a variable anymore.

Reply:
wym?

Reply:
Could be removed really, since it might as well be checking if list is set. But that's not really a problem with this PR.